src/worker/main.ts

```diff
@@ -127,6 +128,22 @@ const sentryMiddleware = (jobHandler) => async (job) => {
 	}
 };
 
+const setDelayOnRateLimiting = (jobHandler) => async (job) => {
+	try {
+		await jobHandler(job);
+	} catch (err) {
+		if (err instanceof RateLimitingError) {
+			const delay = err.rateLimitReset * 1000 - new Date().getTime();
+			logger.warn({ job }, `Rate limiting detected, delaying job by ${delay} ms`)
+			job.opts.backoff = {
```

Comment:
This is the correct way to set the delay on a job?  I'm not certain that this saves it to the queue properly...

Reply:
Also, if you're doing your own retry logic, I'd probably remove the default backoff strategy and attempts and do it all here.  However, since you can't set the number of attempts made, you'll have to add it to the `job.data` property, unless you want to make it that it's impossible for it to ever fail.

Reply:
it's not a nice way of doing it, but it works, and maybe it's even the intended way ...